core/common/src/main/java/alluxio/security/authorization/ExtendedACLEntries.java

```diff
@@ -206,6 +206,38 @@ public AclActions getMask() {
     return actions;
   }
 
+  /**
+   * Update the mask to be the union of owning group entry, named user entry and named group entry.
+   * @param groupActions the group entry to be integrated into the mask
```

Comment:
why does the entire mask have to be recomputed, when it sounds like we just want to update it with the new group actions?

Reply:
the entire mask needs to be recomputed in the case of a removal of entry. 
I updated the code so that the mask is recomputed once for each command, rather than each addition / removal.

Reply:
Can you comment that this must be called when the entries are modified? (And maybe also add comments to setEntries() to call this after changing ACLs)

Reply:
done